go/apps/mixnet/conn.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixnet
+
+import (
+	"errors"
+	"net"
+)
+
+const CellBytes = 1024 // Length of a cell
+
+var errCellLength error = errors.New("incorrect cell length")
+
+// Conn implements the net.Conn interface. The read and write operations are
+// overloaded to check that only cells are sent between agents in the mixnet
+// protocol.
+type Conn struct {
+	net.Conn
+}
+
+// Read a cell from the channel. Return an error if the length
+func (c Conn) Read(msg []byte) (n int, err error) {
+	if n, err = c.Conn.Read(msg); err != nil {
+		return n, err
+	} else if n != CellBytes {
```

Comment:
That wouldn't work because n would be out of scope in the last return. I would have to write:
n, err = c.Conn.Read(msg)
if err != nil {
return n, err
} else if n != CellBytes {
return n, errCellLength
}
return n, nil